### PR TITLE
Improve test coverage for Application

### PR DESCRIPTION
**Pull request by stohn777**
This change is [Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/at4dx/41)

### CODE REVIEW

File: sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES 
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL 
+ *  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ *  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ *  OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**/
+
+public inherited sharing class TestSelectorMethod
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls:27]: @stohn777, please move this class to be an inner class of `ApplicationSObjectSelectorTest`.  Just make the `TestSelectorMethod.Parameters` class as a inner class of `ApplicationSObjectSelectorTest` as well.

File: sfdx-source/core/test/classes/framework-application-factory/TestOnlyAccounts.cls [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES 
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL 
+ *  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ *  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ *  OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**/
+
+public with sharing class TestOnlyAccounts
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-application-factory/TestOnlyAccounts.cls:27]: @stohn777, please move `TestOnlyAccounts` to be a inner class of `ApplicationFactoryTest` and move the `Constructor` and `ITestOnlyAccounts` as direct inner classes of `ApplicationFactoryTest`.

File: sfdx-source/core/test/classes/framework-application-factory/TestOnlyClasses.cls [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL 
+ *  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ *  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ *  OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**/
+
+@IsTest
+public class TestOnlyClasses
```
Comment (ImJohnMDaniel) [inline sfdx-source/core/test/classes/framework-application-factory/TestOnlyClasses.cls:28]: @stohn777, please rename this to be `ApplicationFactoryTestOnlyClasses`

**Review by ImJohnMDaniel — CHANGES_REQUESTED** [review]

File: sfdx-source/core/main/classes/framework-application-factory/Application.cls [hunk 4]
```diff
@@ -215,10 +223,19 @@ public class Application
...
-            // Construct Domain class passing in the queried records
-            fflib_SObjectDomain.IConstructable2 domainConstructor = (fflib_SObjectDomain.IConstructable2) di_Injector.Org.getInstance( IApplicationSObjectDomain.class, domainSObjectType );
+            Object injector_return = di_Injector.Org.getInstance( IApplicationSObjectDomain.class, domainSObjectType );
+
+            if (injector_return instanceOf fflib_SObjectDomain.IConstructable2)
+            {
+                return (IApplicationSObjectDomain) ((fflib_SObjectDomain.IConstructable2) injector_return).construct(records, domainSObjectType);
+            }
+
+            if (injector_return instanceOf IApplicationSObjectDomain)
```
[the remark this answers is not in the document]
Comment (stohn777) [reply sfdx-source/core/main/classes/framework-application-factory/Application.cls:233]: During stub-based mocking.  This is the case that Mr. Chaplin raised where a stub-based mock domain cannot have be an instance of `IConstructable2`.

File: sfdx-source/core/test/classes/framework-application-factory/TestOnlyAccounts.cls (hunk 2 above)
Comment (stohn777) [reply sfdx-source/core/test/classes/framework-application-factory/TestOnlyAccounts.cls:27]: Illegal in Apex with an inner class having an `implements`.  Something to that effect; can't remember the compile error verbatim.

File: sfdx-source/core/test/classes/framework-application-factory/TestOnlyClasses.cls (hunk 3 above)
Comment (stohn777) [reply sfdx-source/core/test/classes/framework-application-factory/TestOnlyClasses.cls:28]: Agreed.

File: sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls (hunk 1 above)
Comment (stohn777) [reply sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls:27]: Illegal in Apex with an inner class having an `implements`.  Something to that effect; can't remember the compile error verbatim.

File: sfdx-source/core/test/classes/framework-platform-event-distributor/TestPlatformEventsConsumer.cls [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES 
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL 
+ *  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ *  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ *  OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**/
+
+public class TestPlatformEventsConsumer
```
[the remark this answers is not in the document]
Comment (stohn777) [reply sfdx-source/core/test/classes/framework-platform-event-distributor/TestPlatformEventsConsumer.cls:27]: Although this has only one class in it, the pattern is consistent with `TestOnlyClasses` or soon to be `ApplicationFactoryTestOnlyClasses`.

File: sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls (hunk 1 above)
Comment (stohn777) [reply sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls:27]: Not `implements`.  By putting this class in the other, then we have 3 levels of classes which is illegal in Apex.

File: sfdx-source/core/test/classes/framework-application-factory/TestOnlyAccounts.cls (hunk 2 above)
Comment (stohn777) [reply sfdx-source/core/test/classes/framework-application-factory/TestOnlyAccounts.cls:27]: Not `implements`.  By putting this class in the other, then we have 3 levels of classes which is illegal in Apex.

Comment (ImJohnMDaniel) [reply sfdx-source/core/test/classes/framework-application-factory/TestOnlyAccounts.cls:27]: @stohn777 You would just move the third-tier inner classes as independent second-tier inner classes.

File: sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls (hunk 1 above)
Comment (ImJohnMDaniel) [reply sfdx-source/core/test/classes/framework-application-factory/TestSelectorMethod.cls:27]: @stohn777 You would just move the third-tier inner classes as independent second-tier inner classes.

File: sfdx-source/core/test/classes/framework-platform-event-distributor/TestPlatformEventsConsumer.cls (hunk 5 above)
[the remark this answers is not in the document]
Comment (ImJohnMDaniel) [reply sfdx-source/core/test/classes/framework-platform-event-distributor/TestPlatformEventsConsumer.cls:27]: I hear ya but I am trying to clutter  up the framework with independent "test" classes.  Better to hide them in as inner classes

**Review by stohn777 — COMMENTED** [review]
@ImJohnMDaniel   The method at `Application.cls` line 320 appears unused and deleting it would increase our code coverage, in addition to removing dead code.  Can you recall anything about it?

**Comment by ImJohnMDaniel** [comment]
> @ImJohnMDaniel The method at `Application.cls` line 320 appears unused and deleting it would increase our code coverage, in addition to removing dead code. Can you recall anything about it?

@stohn777 -- I think that you are right and that it is not need.  I am removing it now.

**Comment by ImJohnMDaniel** [comment]
> @ImJohnMDaniel The method at `Application.cls` line 320 appears unused and deleting it would increase our code coverage, in addition to removing dead code. Can you recall anything about it?

@stohn777 -- specifically which method are you referring to?

**Comment by stohn777** [comment]
> 
> 
> > @ImJohnMDaniel The method at `Application.cls` line 320 appears unused and deleting it would increase our code coverage, in addition to removing dead code. Can you recall anything about it?
> 
> @stohn777 -- specifically which method are you referring to?

Hmm?  The line number must've been from a different version.  This method ...
`private static Type determineClassType( Object instantiatedClass )`

File: sfdx-source/core/main/classes/framework-application-factory/Application.cls (hunk 4 above)
[the remark this answers is not in the document]
Comment (stohn777) [reply sfdx-source/core/main/classes/framework-application-factory/Application.cls:233]: Referring to this issue.
https://github.com/apex-enterprise-patterns/at4dx/issues/25

**Review by ImJohnMDaniel — APPROVED** [review]

**Pull request merged**